File: swift/test/Ice/operations/Oneways.swift
// Copyright (c) ZeroC, Inc.

import Ice
import TestCommon

func oneways(_ helper: TestHelper, _ prx: MyClassPrx) async throws {
    let p = prx.ice_oneway()

    try await p.ice_ping()
    try await p.opVoid()
    try await p.opIdempotent()

    // Calling a ["oneway"] operation on a oneway proxy succeeds.
    try await p.opOneway()

    // Calling a ["oneway"] operation on a twoway proxy throws OnewayOnlyException.
    do {
        try await prx.ice_twoway().opOneway()
        try helper.test(false)
    } catch is Ice.OnewayOnlyException {}

    do {
        _ = try await p.opByte(p1: 0xFF, p2: 0x0F)
        try helper.test(false)
    } catch is Ice.TwowayOnlyException {}
}

File: ruby/test/Ice/operations/Oneways.rb
# Copyright (c) ZeroC, Inc.

require 'Ice'

def oneways(helper, p)
    p = Test::MyClassPrx.uncheckedCast(p.ice_oneway())

    # Calling a ["oneway"] operation on a oneway proxy succeeds.
    p.opOneway()

    # Calling a ["oneway"] operation on a twoway proxy throws OnewayOnlyException.
    begin
        Test::MyClassPrx.uncheckedCast(p.ice_twoway()).opOneway()
        test(false)
    rescue Ice::OnewayOnlyException
    end
end
